valinta-tulos-service/src/main/scala/fi/vm/sade/valintatulosservice/vastaanottomeili/EmailerServlet.scala

```diff
@@ -0,0 +1,122 @@
+package fi.vm.sade.valintatulosservice.vastaanottomeili
+
+import fi.vm.sade.utils.slf4j.Logging
+import fi.vm.sade.valintatulosservice.valintarekisteri.db.SessionRepository
+import fi.vm.sade.valintatulosservice.valintarekisteri.domain.{HakemusOid, HakuOid, HakukohdeOid, ValintatapajonoOid}
+import fi.vm.sade.valintatulosservice.{CasAuthenticatedServlet, VtsServletBase, VtsSwaggerBase}
+import org.scalatra.swagger.SwaggerSupportSyntax.OperationBuilder
+import org.scalatra.swagger.{SwaggerSupport, _}
+import org.scalatra.{InternalServerError, Ok}
+
+import scala.util.{Failure, Success}
+
+
+class EmailerServlet(emailerService: EmailerService, val sessionRepository: SessionRepository)(implicit val swagger: Swagger)
+  extends VtsServletBase with CasAuthenticatedServlet with EmailerSwagger with Logging {
+
+  post("/run", operation(postRunEmailerSwagger)) {
+    logger.info("EmailerServlet POST /run called")
+    emailerService.run() match {
+      case Success(ids) =>
+        Ok(ids)
+      case Failure(e) =>
+        logger.error("Failed to send email: ", e)
+        InternalServerError(e)
+    }
+  }
+
+  post("/run/haku/:hakuOid", operation(postRunEmailerForHakuSwagger)) {
+    val hakuOid = HakuOid(params("hakuOid"))
+    logger.info(s"EmailerServlet POST /run/haku/ called for haku $hakuOid")
+    emailerService.runForHaku(hakuOid) match {
+      case Success(ids) =>
+        Ok(ids)
+      case Failure(e) =>
+        logger.error(s"Failed to send email for haku $hakuOid: ", e)
+        InternalServerError(e)
+    }
+  }
+
+  post("/run/hakukohde/:hakukohdeOid", operation(postRunEmailerForHakukohdeSwagger)) {
+    val hakukohdeOid = HakukohdeOid(params("hakukohdeOid"))
+    logger.info(s"EmailerServlet POST /run/hakukohde/ called for hakukohde $hakukohdeOid")
+    emailerService.runForHakukohde(hakukohdeOid) match {
+      case Success(ids) =>
+        Ok(ids)
+      case Failure(e) =>
+        logger.error(s"Failed to send email for hakukohde $hakukohdeOid: ", e)
+        InternalServerError(e)
+    }
+  }
+
+  post("/run/hakukohde/:hakukohdeOid/valintatapajono/:jonoOid", operation(postRunEmailerForValintatapajonoSwagger)) {
```

Comment:
Ja täällä on sama toiste. Jos tämän koko homman refaktoroisi käyttämään vaikka jotain [command patternia](https://pavelfatin.com/design-patterns-in-scala/#command) niin voisi olla vähemmän toistetta. Eli samaan tyyliin kuin `MailerComponent.Query`.